### PR TITLE
RELATED: RAIL-650 throw new ApiError()

### PR DESCRIPTION
**Pull request by zbycz**

### CODE REVIEW

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://ci.intgdc.com/job/gooddata-js-opensource-repository-check/176/ : SUCCESS in 15s
- https://ci.intgdc.com/job/gooddata-js-unit-tests-zuul-docker-el7/141/ : FAILURE in 1m 25s

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://ci.intgdc.com/job/gooddata-js-opensource-repository-check/177/ : SUCCESS in 8s
- https://ci.intgdc.com/job/gooddata-js-unit-tests-zuul-docker-el7/142/ : FAILURE in 1m 09s

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://ci.intgdc.com/job/gooddata-js-opensource-repository-check/178/ : SUCCESS in 11s
- https://ci.intgdc.com/job/gooddata-js-unit-tests-zuul-docker-el7/143/ : FAILURE in 1m 06s

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://ci.intgdc.com/job/gooddata-js-opensource-repository-check/179/ : SUCCESS in 8s
- https://ci.intgdc.com/job/gooddata-js-unit-tests-zuul-docker-el7/144/ : FAILURE in 2m 10s

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://ci.intgdc.com/job/gooddata-js-opensource-repository-check/180/ : SUCCESS in 13s
- https://ci.intgdc.com/job/gooddata-js-unit-tests-zuul-docker-el7/145/ : FAILURE in 3m 27s

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://ci.intgdc.com/job/gooddata-js-opensource-repository-check/181/ : SUCCESS in 9s
- https://ci.intgdc.com/job/gooddata-js-unit-tests-zuul-docker-el7/146/ : FAILURE in 46s

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://ci.intgdc.com/job/gooddata-js-opensource-repository-check/182/ : SUCCESS in 11s
- https://ci.intgdc.com/job/gooddata-js-unit-tests-zuul-docker-el7/147/ : FAILURE in 52s

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://ci.intgdc.com/job/gooddata-js-opensource-repository-check/185/ : SUCCESS in 8s
- https://ci.intgdc.com/job/gooddata-js-unit-tests-zuul-docker-el7/150/ : FAILURE in 29s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://ci.intgdc.com/job/gooddata-js-opensource-repository-check/186/ : SUCCESS in 11s
- https://ci.intgdc.com/job/gooddata-js-unit-tests-zuul-docker-el7/151/ : SUCCESS in 47s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://ci.intgdc.com/job/gooddata-js-opensource-repository-check/187/ : SUCCESS in 8s
- https://ci.intgdc.com/job/gooddata-js-unit-tests-zuul-docker-el7/152/ : SUCCESS in 1m 49s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://ci.intgdc.com/job/gooddata-js-opensource-repository-check/188/ : SUCCESS in 8s
- https://ci.intgdc.com/job/gooddata-js-unit-tests-zuul-docker-el7/153/ : SUCCESS in 46s

File: src/xhr.js [hunk 1]
```diff
@@ -155,96 +164,104 @@ export function createModule(configStorage) {
...
-     * @method get
-     */
-    const get = (url, settings) => {
-        const opts = merge({ method: 'GET' }, settings);
-        return ajax(url, opts).then(parseJSON);
-    };
+            return handlePolling(finalUrl, finalSettings, ajax);
+        }
+
+        if (response.status >= 200 && response.status <= 399) {
```
Comment (martinmosko) [inline src/xhr.js]: btw correct condition formatting is:

**Comment by zbycz** [comment]
Will be merged with typescript PR.

**Pull request closed without merging**